addons/iap/models/iap.py

```diff
@@ -171,12 +171,21 @@ def get(self, service_name, force_create=True):
             order='id desc')
         if not accounts:
             if force_create:
-                account = self.create({'service_name': service_name})
-                # Since the account did not exist yet, we will encounter a NoCreditError,
-                # which is going to rollback the database and undo the account creation,
-                # preventing the process to continue any further.
-                self.env.cr.commit()
-                return account
+                with self.pool.cursor() as cr:
+                    # Since the account did not exist yet, we will encounter a NoCreditError,
+                    # which is going to rollback the database and undo the account creation,
+                    # preventing the process to continue any further.
+                    IapAccount = self.with_env(self.env(cr=cr))
+                    account = IapAccount.search([
+                        ('service_name', '=', service_name),
+                        '|',
+                            ('company_ids', 'in', self.env.companies.ids),
+                            ('company_ids', '=', False)],
```

Comment:
For the sake of code readability, I suggest to define `domain` with this value at the beginning of method `get`.